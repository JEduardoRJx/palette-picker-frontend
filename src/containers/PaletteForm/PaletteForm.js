import React, { Component } from 'react';
import '../../variables.scss'
import './PaletteForm.scss';

export class PaletteForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      currentPalette: '',
      colors: [],
      error: ''
    }
  }

  handleChange = (e) => {
    this.setState({error: ''})
    let currentPalette = this.state.currentPalette;
    currentPalette = { ...currentPalette, [e.target.name]: e.target.value};
    this.setState({currentPalette: currentPalette})
  }

  render() {
    console.log("props", this.props)
    const { palettes, currentProjectId } = this.props;
    console.log("pale", typeof parseInt(currentProjectId))
    console.log("allPalettes", palettes.flat())
    const paletteNames = palettes.flat().map(currentPalette => {
      console.log("currentP", currentPalette)
      if (currentPalette.project_id === parseInt(currentProjectId)) {
        console.log("truuuuue")
        return <option value={currentPalette.palette_name} key={currentPalette.id}> {currentPalette.palette_name} </option>
      } else {
        return
      }
    })
    return (
      <section className="palette-details">
        <form className="palette-form">
          <input 
            type="text" 
            name="palette_name" 
            placeholder="Enter Palette Name"
            value = {this.state.name}
            onChange = {this.handleChange}
          />
        </form>
        <select className="palette-select">
          <option value="Palette Name"> Select a Palette </option>
          {paletteNames}
          {/* <option value="Palette Name 1"> Palette 1 </option> */}
          {/* <option value="Palette Name 2"> Palette 2 </option> */}
        </select>
      </section>
    )
  }
}

export default PaletteForm;
