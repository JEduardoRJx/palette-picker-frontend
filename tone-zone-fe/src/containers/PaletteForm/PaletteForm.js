import React, { Component } from 'react';

export class PaletteForm extends Component {
  constructor() {
    super()
    this.state = {
      currentPalette: {
        name: '',
        colors: []
      },
      allPalettes: [],
      error: ''
    }
  }
  render() {
    return (
      <section className="palette-details">
        <form className="palette-form">
          Enter Palette Name: <input type="text" name="projectName"></input>
        <input type="submit"></input>
        </form>
        <select>
          <option value="Project Name"> Project Name </option>
          <option value="Project Name 1"> Project 1 </option>
          <option value="Project Name 2"> Project 2 </option>
        </select>
      </section>
    )
  }
}

export default PaletteForm;
