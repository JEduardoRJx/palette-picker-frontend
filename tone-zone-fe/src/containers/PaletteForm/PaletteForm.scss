.palette-details {
  width: 33.3%;
}
