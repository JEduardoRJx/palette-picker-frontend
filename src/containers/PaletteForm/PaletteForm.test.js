import React from 'react';
import { shallow } from 'enzyme';
import { PaletteForm } from './PaletteForm'

describe('PaletteForm ', () => {
  it('should match the snapshot ', () => {
    const wrapper = shallow(<PaletteForm  />);
    expect(wrapper).toMatchSnapshot();
  });
});