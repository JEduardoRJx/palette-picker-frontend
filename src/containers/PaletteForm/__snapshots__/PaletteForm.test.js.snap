// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteForm  should match the snapshot  1`] = `
<section
  className="palette-details"
>
  <form
    className="palette-form"
  >
    <input
      name="palette_name"
      onChange={[Function]}
      placeholder="Enter Palette Name"
      type="text"
    />
  </form>
  <select
    className="palette-select"
  >
    <option
      value="Palette Name"
    >
       Select a Palette 
    </option>
    <option
      value="Palette Name 1"
    >
       Palette 1 
    </option>
    <option
      value="Palette Name 2"
    >
       Palette 2 
    </option>
  </select>
</section>
`;
